matplotlib_pyodide/html5_canvas_backend.py

```diff
@@ -204,8 +204,8 @@ def __init__(self, ctx, width, height, dpi, fig):
         self.ctx.width = self.width
         self.ctx.height = self.height
         self.dpi = dpi
-        self.fontd = maxdict(50)
         self.mathtext_parser = MathTextParser("bitmap")
+        self._get_font_helper = lru_cache(maxsize=50)(self._get_font_helper)
```

Comment:
What's the difference between adding a `lru_cache` decorator on top of `self._get_font_helper`, or re-defining a variable like this?

Reply:
Two things:
1. Decorating the method makes one shared cache across all class instances, this makes a per-instance cache. It actually might make more sense to have a shared cache here but I don't understand what's going on well enough to know.
2. The cache holds a strong reference to the arguments of the function including the`self` argument so if you decorate the method directly then instances can be kept alive by the cache. This instead makes a reference loop but the gc will collect it.


It would probably be better to shift this to a top level or static method and use a shared cache.

Reply:
I guess the main concern is that the font objects appear to be mutable. I wonder if there is a good way to copy them so we can avoid mutating cache entries...

Reply:
I see. Thanks for the explanation!